pkg/controller/v1alpha2/experiment/util/client_util.go

```diff
@@ -16,14 +16,118 @@ limitations under the License.
 package util
 
 import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	experimentv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	cl "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
-func NewClient(options client.Options) (client.Client, error) {
+type katibClient struct {
+	client client.Client
+}
+
+const (
+	trialTemplatesName            = "trial-template"
+	metricsCollectorTemplatesName = "metricscollector-template"
```

Comment:
How about adding it in pkg/api/operators/apis/experiment/v1alpha2/constants.go?  trialTemplate is already defined